### PR TITLE
feat: Implement _mm_undefined_pd

### PR DESCRIPTION
**Pull request by HowJMay**

### CODE REVIEW

**Comment by jserv** [comment]
Did you implement the corresponding test case?

File: tests/impl.cpp [hunk 1]
```diff
@@ -6280,7 +6280,9 @@ result_t test_mm_ucomineq_sd(const SSE2NEONTestImpl &impl, uint32_t iter)
 
 result_t test_mm_undefined_pd(const SSE2NEONTestImpl &impl, uint32_t iter)
 {
-    return TEST_UNIMPL;
+    __m128d a = _mm_undefined_pd();
+    (void) a;
```
Comment (marktwtn) [inline tests/impl.cpp:6284]: What is this for?

Comment (HowJMay) [reply tests/impl.cpp:6284]: To remove the unused variable warning

Comment (marktwtn) [reply tests/impl.cpp:6284]: I guess we might just return `TEST_SUCESS`
or replace the code with `_mm_undefined_pd()` without declaring variable.

@jserv any suggestion?

Comment (jserv) [reply tests/impl.cpp:6284]: We can rewrite as following:
```c
__m128d a = _mm_undefined_pd();
a = _mm_xor_pd(a, a);
return validateDouble(a, 0, 0);
```

**Review by marktwtn — APPROVED** [review]

**Pull request merged**